src/ddmd/astbase.d

```diff
@@ -0,0 +1,6299 @@
+module ddmd.astbase;
+
+import ddmd.astbasevisitor;
+
+struct ASTBase
+{
+    import ddmd.root.file;
+    import ddmd.root.filename;
+    import ddmd.root.array;
+    import ddmd.root.rootobject;
+    import ddmd.root.outbuffer;
+    import ddmd.root.ctfloat;
+    import ddmd.root.rmem;
+    import ddmd.root.stringtable;
+
+    import ddmd.tokens;
+    import ddmd.identifier;
+    import ddmd.globals;
+    import ddmd.id;
+    import ddmd.errors;
+    import ddmd.lexer;
+
+    import core.stdc.string;
+    import core.stdc.stdarg;
+
+    alias Dsymbols              = Array!(Dsymbol);
```

Comment:
I took inspiration from src/ddmd/arraytypes.d. I wanted to have consistency with the rest of the compiler code.